integration/validation_separate_metrics_test.go

```diff
@@ -0,0 +1,296 @@
+	})
+
+	res, err = client.Push(series2)
+	require.NoError(t, err)
+	require.Equal(t, 400, res.StatusCode)
+
+	metricNumSeries, err = distributor.SumMetrics([]string{"cortex_discarded_samples_total"},
+		e2e.WithLabelMatchers(labels.MustNewMatcher(labels.MatchEqual, "group", "")),
+		e2e.WaitMissingMetrics)
+
+	require.NoError(t, err)
+	require.Equal(t, 1, len(metricNumSeries))
+	require.Equal(t, float64(2), metricNumSeries[0])
+
+	metricNumSeries, err = distributor.SumMetrics([]string{"cortex_discarded_samples_total"},
+		e2e.WithLabelMatchers(labels.MustNewMatcher(labels.MatchEqual, "group", "second-series")),
+		e2e.WaitMissingMetrics)
+
+	require.NoError(t, err)
+	require.Equal(t, 1, len(metricNumSeries))
+	require.Equal(t, float64(1), metricNumSeries[0])
+}
+
+func TestPushMultipleInvalidLabels(t *testing.T) {
+
+	s, err := e2e.NewScenario(networkName)
+	require.NoError(t, err)
+	defer s.Close()
+
+	flags := mergeFlags(
+		BlocksStorageFlags(),
+		BlocksStorageS3Flags(),
+	)
+
+	flags["-validation.separate-metrics-group-label"] = "separate_metrics_group"
+
+	// Start dependencies.
+	consul := e2edb.NewConsul()
+	minio := e2edb.NewMinio(9000, flags["-blocks-storage.s3.bucket-name"])
+	require.NoError(t, s.StartAndWaitReady(consul, minio))
+
+	// Start Mimir components.
+	distributor := e2emimir.NewDistributor("distributor", consul.NetworkHTTPEndpoint(), flags)
+	ingester1 := e2emimir.NewIngester("ingester-1", consul.NetworkHTTPEndpoint(), flags)
+	ingester2 := e2emimir.NewIngester("ingester-2", consul.NetworkHTTPEndpoint(), flags)
+	ingester3 := e2emimir.NewIngester("ingester-3", consul.NetworkHTTPEndpoint(), flags)
+	require.NoError(t, s.StartAndWaitReady(distributor, ingester1, ingester2, ingester3))
+
+	// Wait until distributor has updated the ring.
+	require.NoError(t, distributor.WaitSumMetricsWithOptions(e2e.Equals(3), []string{"cortex_ring_members"}, e2e.WithLabelMatchers(
+		labels.MustNewMatcher(labels.MatchEqual, "name", "ingester"),
+		labels.MustNewMatcher(labels.MatchEqual, "state", "ACTIVE"))))
+
+	// Wait until ingesters have heartbeated the ring after all ingesters were active,
+	// in order to update the number of instances. Since we have no metric, we have to
+	// rely on a ugly sleep.
+	time.Sleep(2 * time.Second)
+
+	client, err := e2emimir.NewClient(distributor.HTTPEndpoint(), "", "", "", userID)
+	require.NoError(t, err)
+
+	// Push invalid series with different groups
+	for i := 0; i < 10; i++ {
+		series, _, _ := generateSeries("TestMetric", time.Now(), prompb.Label{
+			Name:  "separate_metrics_group",
+			Value: strconv.Itoa(i % 2),
+		}, prompb.Label{
+			Name:  "Test|Invalid|Label|Char",
+			Value: "123",
+		})
+
+		res, err := client.Push(series)
+		require.NoError(t, err)
+		require.Equal(t, 400, res.StatusCode)
+	}
+
+	metricNumSeriesOdd, err := distributor.SumMetrics([]string{"cortex_discarded_samples_total"},
+		e2e.WithLabelMatchers(labels.MustNewMatcher(labels.MatchEqual, "group", "1")),
```

Comment:
You're both asserting on the group `1` here and below. Change the `for` loop to iterate over `9` instead of `10` and the test should fail.